base/int.jl

```diff
@@ -31,8 +31,8 @@ const BitUnsigned64T   = Union{Type{UInt8}, Type{UInt16}, Type{UInt32}, Type{UIn
 
 const BitIntegerType = Union{map(T->Type{T}, BitInteger_types)...}
 
-throw_inexacterror(f::Symbol, ::Type{T}, val) where T =
-    (@_noinline_meta; throw(InexactError(f, T, val)))
+# calls constructors defined in boot.jl
+convert(T::BitIntegerType, x::Union{BitInteger, Bool}) = T(x)
```

Comment:
in abstract_call_method, there's a heuristic that should be changed to reflect the new signature:
```julia
        # the construct-to-convert method is a bottleneck in inference,
        # so just assume that recursion will get prevented at some other point
       && !(method.sig == Tuple{Type, Any}))
```
that should address the inference failure in mmap